### PR TITLE
IndexShard state should also be started.

### LINKED ISSUE #38617: [CI] IndexFollowingIT.testIndexFallBehind some times failes on 6.7
Build failure: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+6.7+internalClusterTest/137/console

Reproduce with: 

```
./gradlew :x-pack:plugin:ccr:internalClusterTest -Dtests.seed=8A5DBECB542A869B -Dtests.class=org.elasticsearch.xpack.ccr.IndexFollowingIT -Dtests.method="testIndexFallBehind" -Dtests.security.manager=true -Dtests.locale=es-CU -Dtests.timezone=America/Vancouver -Dcompiler.java=11 -Druntime.java=8
```

Failure:

```
ERROR   0.59s J7 | IndexFollowingIT.testIndexFallBehind <<< FAILURES!
14:04:13    > Throwable #1: org.apache.lucene.store.AlreadyClosedException: engine is closed
14:04:13    > 	at __randomizedtesting.SeedInfo.seed([8A5DBECB542A869B:6DB7F91E3A82A6DD]:0)
14:04:13    > 	at org.elasticsearch.index.shard.IndexShard.getEngine(IndexShard.java:2264)
14:04:13    > 	at org.elasticsearch.index.shard.IndexShardTestCase.getDocIdAndSeqNos(IndexShardTestCase.java:698)
14:04:13    > 	at org.elasticsearch.xpack.CcrIntegTestCase.getDocIdAndSeqNos(CcrIntegTestCase.java:490)
14:04:13    > 	at org.elasticsearch.xpack.CcrIntegTestCase.lambda$assertIndexFullyReplicatedToFollower$4(CcrIntegTestCase.java:454)
14:04:13    > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:836)
14:04:13    > 	at org.elasticsearch.xpack.CcrIntegTestCase.assertIndexFullyReplicatedToFollower(CcrIntegTestCase.java:453)
14:04:13    > 	at org.elasticsearch.xpack.ccr.IndexFollowingIT.testIndexFallBehind(IndexFollowingIT.java:997)
14:04:13    > 	at java.lang.Thread.run(Thread.java:748)
```

I managed to reproduce this locally by running:

```
./gradlew :x-pack:plugin:ccr:internalClusterTest -Dtests.class=org.elasticsearch.xpack.ccr.IndexFollowingIT -Dtests.method="testIndexFallBehind" -Dtests.iters=20
```

It failed once in my first try.

### PR DESCRIPTION
**Pull request by martijnvg**
Fixes #38617 (the test also failed in 8.0 / 7.x, but happens less likely)

### CODE REVIEW

**Comment by martijnvg** [comment]
Note this change should be backported to 7.x, 7.0 and 6.7 branches. The `IndexFollowingIT#testIndexFallBehind()` test has been muted on 7.0 and 6.7 branches, because it failed too often there.

File: x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/CcrIntegTestCase.java [hunk 1]
```diff
@@ -487,6 +488,9 @@ protected void assertIndexFullyReplicatedToFollower(String leaderIndex, String f
             }
             IndexShard indexShard = cluster.getInstance(IndicesService.class, state.nodes().get(shardRouting.currentNodeId()).getName())
                 .indexServiceSafe(shardRouting.index()).getShard(shardRouting.id());
+            if (indexShard.state() != IndexShardState.STARTED) {
+                continue;
+            }
```
Comment (dnhatn) [inline x-pack/plugin/ccr/src/test/java/org/elasticsearch/xpack/CcrIntegTestCase.java]: I think we should catch "AlreadyClosedException" when calling `IndexShardTestCase.getDocIdAndSeqNos(indexShard)` instead?

**Review by dnhatn — COMMENTED** [review]
Thanks @martijnvg. I left a comment.

**Comment by martijnvg** [comment]
@dnhatn I've updated the PR.

**Review by dnhatn — APPROVED** [review]
LGTM. Thanks @martijnvg.

**Comment by jasontedor** [comment]
@martijnvg Does this need to be backported to 6.7?

**Comment by martijnvg** [comment]
@jasontedor Yes, and it was backported: https://github.com/elastic/elasticsearch/commit/232d2e0a21ddc4e7175ae6994c16471140375c97

**Comment by martijnvg** [comment]
I just forgot to update the version labels here.

**Comment by jasontedor** [comment]
Okay, I think that I see what happened. The test is still marked as awaits fix in the 6.7 and 7.0 branch. I think that is the case because they were never marked as awaits fix in master, so that when you developed the fix you didn't have to remove the awaits fix, and hence no removal of the awaits fix when you cherry picked back.

**Pull request merged**